src/kernels/kernelAutoReConnectFailedMonitor.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import type { Kernel } from '@jupyterlab/services';
+import { inject, injectable } from 'inversify';
+import { NotebookCell } from 'vscode';
+import { IExtensionSyncActivationService } from '../platform/activation/types';
+import { IDisposableRegistry } from '../platform/common/types';
+import { DataScience } from '../platform/common/utils/localize';
+import { Telemetry } from '../telemetry';
+import { endCellAndDisplayErrorsInCell } from './execution/helpers';
+import { getDisplayNameOrNameOfKernelConnection } from './helpers';
+import { sendKernelTelemetryEvent } from './telemetry/sendKernelTelemetryEvent';
+import { IKernel, IKernelProvider, isLocalConnection } from './types';
+
+/**
+ * In the case of Jupyter kernels, when a kernel dies Jupyter will automatically restart that kernel.
+ * In such a case we need to display a little progress indicator so user is aware of the fact that the kernel is restarting.
+ */
+@injectable()
+export class kernelAutoReConnectFailedMonitor implements IExtensionSyncActivationService {
```

Comment:
Minor, but I think we capitalize all our class names, at least I can't think of any exceptions offhand.

Reply:
oops, oh yes agreed, that's a mistake.